### PR TITLE
Social media block #122

### PR DESCRIPTION
**Pull request by UcDust**

### CODE REVIEW

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      </${BlockControls}>
+      ${!attributes.facebookUrl && !attributes.twitterUrl && !attributes.instagramUrl && !attributes.youtubeUrl && !attributes.linkedinUrl && html`
+        <div>
+          <div style=${{backgroundColor: '#1da1f2', width: '40px', height: '40px', display: 'inline-block'}}></div>
+          <div style=${{backgroundColor: '#1778f2', width: '40px', height: '40px', display: 'inline-block'}}></div>
+          <div style=${{backgroundColor: '#f00075', width: '40px', height: '40px', display: 'inline-block'}}></div>
+        </div>
+      `}
+      ${attributes.facebookUrl ? html`
+        <ucdlib-icon icon='ucd-public:facebook' class="media-icon"></ucdlib-icon>
```
Comment (spelkey-ucd) [inline src/editor/lib/blocks/ucd-theme-social-media/edit.js:91]: sorry, i misunderstood your question about using the icons.
don't use `ucdlib-icon`. try to follow the brand styles as closely as possible: 
 - host the [sprite sheet on the site](https://github.com/UCDavisLibrary/ucdlib-theme-wp/tree/sandbox/assets/img). 
 - Then add a css rule that modifies `.social-follow a` to be `background-image:url(/wp-content/themes/ucdlib-theme-wp/assets/img/social-follow.svg)`

does that make sense?

**Review by spelkey-ucd — CHANGES_REQUESTED** [review]

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import { useBlockProps, BlockControls } from '@wordpress/block-editor';
+import { TextControl, ToolbarButton, Button, Modal, URLInput } from '@wordpress/components';
+import { useRef, useState, useEffect } from "@wordpress/element";
+
+export default ( props ) => {
+  const { attributes, setAttributes } = props;
+  const blockProps = useBlockProps();
+  const mainEleRef = useRef();
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (UcDust) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js:11]: We were hitting a problem with the default properties in that if the property isn't set to something other than the default, then it's not saved into the json/db. It's available in the editor/client side, but the values wouldn't be available in the public side. Is there a more elegant way around that?

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js (hunk 1 above)
Comment (UcDust) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js:91]: I think so, but the spritesheet is an svg with all 5 icons. Is there a way to set the background-image to a specific viewport of an svg? 
Cosmetically in the UI they match the brand styles very closely, just technically they're wrapped in `ucdlib-icon` markup.

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (spelkey-ucd) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js:11]: nope. not that i know of. just have to define them twice.

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js (hunk 1 above)
Comment (spelkey-ucd) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js:91]: yeah, the patternlab styles should do it for you if you just set the class: 
image

http://dev.webstyleguide.ucdavis.edu/redesign/?p=molecules-social-follow

File: src/editor/lib/blocks/ucd-theme-social-media/edit.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (UcDust) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js:11]: I'm wondering, if we do check in the backend for an attribute or a default value, what happens if the user has emptied out the attribute in the editor side and doesn't want the link to show up? For instance, if the twig is doing something like this-

` {% if attributes.facebookUrl || 'https://www.facebook.com/UCDavisLibrary/' %}`

[the remark this answers is not in the document]
Comment (spelkey-ucd) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js]: ohh duh. thats a good point. okay, just set the attribute defaults to an empty string, which means nothing is displayed. we can just make a [reusable block](https://wordpress.com/support/wordpress-editor/blocks/reusable-block/) when it is on stage with all of the library's social media.

[the remark this answers is not in the document]
Comment (UcDust) [reply src/editor/lib/blocks/ucd-theme-social-media/edit.js]: That works, ok thanks again! I've committed the latest just now.

**Pull request merged**